rotkehlchen/api/rest.py

```diff
@@ -3046,13 +3047,26 @@ def _decode_given_evm_tx(
         except DeserializationError as e:
             raise InputError(str(e)) from e
 
-        chain_manager.transactions_decoder.decode_and_get_transaction_hashes(
+        events = chain_manager.transactions_decoder.decode_and_get_transaction_hashes(
             tx_hashes=[tx_ref],
             send_ws_notifications=True,
             ignore_cache=True,  # always redecode from here
             delete_customized=delete_custom,
         )
 
+        # Notify the user if a gnosis pay tx was redecoded but the api key is missing.
+        # Otherwise, the event notes are not updated and there's no indication what is wrong.
+        if (
+            any((x.counterparty == CPT_GNOSIS_PAY and x.notes is None) for x in events) and
```

Comment:
lastly can we do better than iterate once again the events? seems that we do it several times already 😓 this solution is also the first one that comes to my mind but not sure if we can do better